### PR TITLE
  [hotfix] FigShare OAuth canellations redirected incorrectly

### PR DESCRIPTION
**Pull request by HarryRybacki**
## Purpose
Ensure redirects for user OAuth requests that are cancelled are handled correctly

## Changes
Cancelled requests from a user's settings page will redirect to their /settings/addons/ and cancelled requests from a projects settings page will be redirected to the project's /settings/ page

## Side Effects
None

Closes-Issue: #1844

### CODE REVIEW

File: website/addons/figshare/tests/test_views.py [hunk 1]
```diff
@@ -23,6 +25,15 @@
 figshare_mock = create_mock_figshare(project=436)
 
 
+def assert_urls_equal(url1, url2):
```
Comment (sloria) [inline website/addons/figshare/tests/test_views.py]: This is already defined in `tests.test_addon` (though it should probably be moved to `tests.base` or `tests.utils`).

File: website/addons/figshare/tests/test_views.py [hunk 2]
```diff
@@ -66,6 +77,24 @@ def test_import_auth(self):
         is_not_none = settings.user_settings != None
         assert_true(is_not_none)
 
+    def test_oauth_cancel_from_user_settings_redirect(self):
+        """Ensures user initiated oauth token requests that are cancelled
```
Comment (sloria) [inline website/addons/figshare/tests/test_views.py]: > Use long, descriptive names. This often obviates the need for doctrings in test methods. This also makes it easier to locate tests that fail.

http://cosdev.readthedocs.org/en/latest/process/testing.html

**Comment by sloria** [comment]
Looks good. Just a few minor changes and this is good to merge. :point_up:

**Comment by HarryRybacki** [comment]
@sloria done!

**Pull request merged**